astroid/typing.py

```diff
@@ -41,4 +41,4 @@ class AstroidManagerBrain(TypedDict):
     _transform: transforms.TransformVisitor
 
 
-InferenceResult = Union["nodes.NodeNG", "type[util.Uninferable]", "bases.Instance"]
+InferenceResult = Union["nodes.NodeNG", "type[util.Uninferable]", "bases.Proxy"]
```

Comment:
Just skimming quickly -- do you mind explaining this part?

Reply:
See L124 of `bases.py`. We initially thought that `Instance` was the "lowest" type of the `bases` that could be returned by `_infer` but it turns out it is `Proxy`.

Any inference in `astroid` can thus either return a node, a proxy or be uninferable.

Reply:
Sounds good. Looks like `Proxy` could be an abstract class someday?

Reply:
> Sounds good. Looks like `Proxy` could be an abstract class someday?

Probably. I don't think we ever instantiate one directly, but I haven't checked.